### PR TITLE
Noam & Hillel - added beam breaker

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java [hunk 1]
```diff
@@ -25,5 +28,6 @@ public void updateInputs(IntakeInputsAutoLogged chassisIntakeInputs) {
 		chassisIntakeInputs.outputCurrent = motor.getOutputCurrent();
 		chassisIntakeInputs.appliedOutput = motor.getAppliedOutput();
 		chassisIntakeInputs.velocity = motor.getEncoder().getVelocity();
+		chassisIntakeInputs.beamBreakerValue = beamBreaker.get();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java:31]: why is the name chassisIntakeIInputs and not just intakeInputs

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java (hunk 1 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java:31]: Thank you Romy for your helpful opinion!!! -noam <3

**Review by bossyshrimp68 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java [hunk 2]
```diff
@@ -2,12 +2,15 @@
 
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.CANSparkMaxLowLevel;
+import edu.wpi.first.wpilibj.DigitalInput;
 
 public class NeoIntake implements IIntake {
 	private CANSparkMax motor;
+	private static DigitalInput beamBreaker;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java:9]: why not debouncer? https://docs.wpilib.org/he/stable/docs/software/advanced-controls/filters/debouncer.html

Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java:9]: and there will be 2 beam breakers: one at the input and one at the end

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/IntakeInputs.java [hunk 3]
```diff
@@ -4,7 +4,9 @@
...
 @AutoLog
 public class IntakeInputs {
-	public static double appliedOutput;
-	public double outputCurrent;
-	public static double velocity;
+    public static double appliedOutput;
+    public static double velocity;
+    public static boolean entranceBeamBreakerValue;
+    public static boolean exitBeamBreakerValue;
+    public double outputCurrent;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/IntakeInputs.java:11]: I want more data

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java [hunk 4]
```diff
@@ -21,9 +26,27 @@ public void setVoltage(double voltage) {
...
+
+
+
+
+
+
+
+
+
 	}
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/NeoIntake.java:50]: Why so many spaces

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/SimulationIntake.java [hunk 5]
```diff
@@ -4,34 +4,51 @@
...
-	public void updateInputs(IntakeInputsAutoLogged chassisIntakeInputs) {
-		chassisIntakeInputs.appliedOutput = appliedOutput;
-		chassisIntakeInputs.outputCurrent = motorSimulation.getCurrentDrawAmps();
-		chassisIntakeInputs.velocity = motorSimulation.getAngularVelocityRPM();
-	}
+    private DCMotorSim motorSimulation;
+    private double appliedOutput;
+
+    private SendableChooser<Boolean> entranceBeamBreaker;
+    private SendableChooser<Boolean> exitBeamBreaker;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/SimulationIntake.java:15]: Tested?

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

**Review by NoamRosenberg08 — APPROVED** [review]

**Review by NoamRosenberg08 — APPROVED** [review]

**Review by itamaroryan — APPROVED** [review]

**Pull request merged**